cts/cts-exec.in

```diff
@@ -151,22 +151,33 @@ class ExecTest(Test):
         self._daemon_process = None
         self._stonith_process = None
 
-    def add_cmd(self, args):
+    # add_cmd, add_cmd_and_kill, add_cmd_check_stdout, and add_cmd_expected_fail
+    # override the superclass versions but deal with the parameters differently.
+    # In this file, the callers do not pass a command but do pass an argument
+    # string as the first parameter.  The command is always the same -
+    # self._test_tool_location.
+    #
+    # In order to avoid changing the many callers, the functions below treat the
+    # first parameter as the argument string, ignore the args parameter, and pass
+    # the superclass versions what they expect.
+
+    def add_cmd(self, cmd=None, args="", validate=True, check_rng=True, check_stderr=True,
+                env=None):
```

Comment:
We're trying to make it less confusing not more ;) I think the only choice will be to update the callers.

This is missing exitcode, and everything past args is ignored (which is good, because they should be False). Maybe something like:

    def add_cmd(self, cmd=None, **kwargs):

        if cmd is None:
            cmd = self._test_tool_location
        self._new_cmd(cmd, **kwargs);